### DISCUSSION TITLE
Audit Report Functionality

### DISCUSSION THREAD

**Discussion by rashmimaharana** [drakkan/sftpgo#537] [Q&A] [answered]
Hi @drakkan and Team,

Is there any way to get audit report data like who uploaded/deleted files. Could see some info in logs. Do we have any api to fetch the details for same.

Thanks for this cool sftp component.

Thanks,
Rashmi Maharana

**Comment by drakkan** [comment]
Hi,

currently these info are recorded in log files. Do you want to sponsor this feature?

**Reply by rashmimaharana** [reply]
Sure @drakkan 
Let me check the best way of doing this and will raise a PR with proper documentation.

**Comment by drakkan** [comment, marked as answer]
> Hi @drakkan and Team,
> 
> Is there any way to get audit report data like who uploaded/deleted files. Could see some info in logs. Do we have any api to fetch the details for same.
> 
> Thanks for this cool sftp component.
> 
> Thanks, Rashmi Maharana

Hi,

I think this can be done using the [eventstore](https://github.com/sftpgo/sftpgo-plugin-eventstore) and [eventsearch](https://github.com/sftpgo/sftpgo-plugin-eventsearch) plugins. Once you configure these plugins you can search events with REST API:

https://github.com/drakkan/sftpgo/blob/31984ffec18e6064947fe2cc109f688f68a7f075/httpd/schema/openapi.yaml#L1489
https://github.com/drakkan/sftpgo/blob/31984ffec18e6064947fe2cc109f688f68a7f075/httpd/schema/openapi.yaml#L1616
